### ISSUE TITLE
fails to build with -fno-common (GCC10 default)

### ISSUE THREAD

**Issue by gyakovlev** [zfsonlinux/zfs#9896]
### System information

Type | Version/Name
 --- | --- 
Distribution Name	| Gentoo
Distribution Version	| rolling
Linux Kernel	| 5.4.14-gentoo
Architecture	| x86_64
ZFS Version	| zfs-0.8.3-r0-gentoo
SPL Version	| zfs-kmod-0.8.3-r0-gentoo

### Describe the problem you're observing
gcc-10 and above flipped a default from `-fcommon` to `-fno-common`

This changed gcc code generator to emit globals without explicit initializer from `.bss` (via `COMMON` symbol type) to `.data` (via `DEFAULT` symbol type). 

zfs fails to build with the following error
```shell-session
...
#	source='../../module/zcommon/zprop_common.c' object='zprop_common.lo' libtool=yes 
/bin/sh ../../libtool  --tag=CC --silent  --mode=compile x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -include ../../zfs_config.h -I../../include -I../../module/icp/include -I../../lib/libspl/include  -D_GNU_SOURCE -D_REENTRANT -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DHAVE_LARGE_STACKS=1 -DTEXT_DOMAIN=\"zfs-linux-user\" -DLIBEXECDIR=\"/usr/libexec\" -DRUNSTATEDIR=\"/var/run\" -DSBINDIR=\"/sbin\" -DSYSCONFDIR=\"/etc\" -UDEBUG -DNDEBUG   -std=gnu99 -Wall -Wstrict-prototypes -fno-strict-aliasing -fno-omit-frame-pointer    -Wno-unused-but-set-variable -O2 -pipe -march=native -fno-common -c -o zprop_common.lo ../../module/zcommon/zprop_common.c
/bin/sh ../../libtool  --tag=CC --silent  --mode=link x86_64-pc-linux-gnu-gcc -std=gnu99 -Wall -Wstrict-prototypes -fno-strict-aliasing -fno-omit-frame-pointer    -Wno-unused-but-set-variable -O2 -pipe -march=native -fno-common -version-info 2:0:0 -Wl,-O1 -Wl,--as-needed -o libzfs.la -rpath /usr/lib64 libzfs_changelist.lo libzfs_config.lo libzfs_crypto.lo libzfs_dataset.lo libzfs_diff.lo libzfs_import.lo libzfs_iter.lo libzfs_mount.lo libzfs_pool.lo libzfs_sendrecv.lo libzfs_status.lo libzfs_util.lo algs/sha2/sha2.lo zfeature_common.lo zfs_comutil.lo zfs_deleg.lo zfs_fletcher.lo zfs_fletcher_aarch64_neon.lo zfs_fletcher_avx512.lo zfs_fletcher_intel.lo zfs_fletcher_sse.lo zfs_fletcher_superscalar.lo zfs_fletcher_superscalar4.lo zfs_namecheck.lo zfs_prop.lo zfs_uio.lo zpool_prop.lo zprop_common.lo ../../lib/libnvpair/libnvpair.la ../../lib/libshare/libshare.la ../../lib/libuutil/libuutil.la ../../lib/libzfs_core/libzfs_core.la ../../lib/libzutil/libzutil.la -lm -lssl -lcrypto -ludev -luuid -lz   
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ../../lib/libshare/.libs/libshare.a(smb.o):(.bss+0x0): multiple definition of `smb_shares'; ../../lib/libshare/.libs/libshare.a(libshare.o):(.bss+0x0): first defined here
collect2: error: ld returned 1 exit status
```

### Describe how to reproduce the problem
Very easy to reproduce on gcc8 or gcc9 by adding `-fno-common` to `CFLAGS`
But since it'll become default in GCC10 we are trying to catch breakages early.

### Include any warning/errors/backtraces from the system logs
full build log
[sys-fs zfs-0.8.3 20200126-185837.log.gz](https://github.com/zfsonlinux/zfs/files/4114325/sys-fs.zfs-0.8.3.20200126-185837.log.gz)

gentoo bug https://bugs.gentoo.org/706534
gcc bug  https://gcc.gnu.org/bugzilla/show_bug.cgi?id=85678

**Comment by trofi** [comment]
Probably covered by https://github.com/zfsonlinux/zfs/commit/af09c050e95bebbaeca52156218f3f91e8c9951a

**Comment by behlendorf** [comment]
@trofi thanks for mentioning this issue.  Yes, this should be resolved by af09c05.
